doc/single-nic-vm-host-with-vlans.md

```diff
@@ -0,0 +1,270 @@
+    - DNS1: ```1.1.1.1```.
+    - DNS2: ```8.8.8.8```.
+- A switch that supports [VLAN](https://en.wikipedia.org/wiki/VLAN).
+- A router/firewall that supports [VLAN](https://en.wikipedia.org/wiki/VLAN).
+    - VLAN1 IPv4: ```192.168.150.254/24```.
+    - VLAN40 IPv4: ```192.168.151.254/24```.
+    - VLAN41 IPv4: ```192.168.152.254/24```.
+    - Firewall policies, interVLAN routing, DNS, DHCP configured and tested.
+
+
+### Elevated privileges
+All commands in this tutorial assume you are using elevated privileges.
+
+### QEMU/KVM installed
+QEMU/KVM should be installed. Installation instructions can be found on [this Ubuntu Wiki page](https://help.ubuntu.com/community/KVM/Installation).
+
+### Disable netfilter for bridged interfaces
+
+We need to disable netfilter for bridged interfaces, in order to allow communications between the homeserver, its virtual machines, and the devices in the local VLANs.
+
+- Edit this file: ```/etc/systemctl.conf```
+- Add the following lines to the file
+
+```
+net.bridge.bridge-nf-call-iptables = 0
+net.bridge.bridge-nf-call-ip6tables = 0
+net.bridge.bridge-nf-call-arptables = 0
+```
+
+- Apply the changes immediately, without rebooting the host.
+
+```
+sysctl -p /etc/sysctl.conf
+```
+
+
+
+### Edit or firewall rules
+
+
+Remember to edit, configure, or disable firewall according to your needs. You can refer to [ufw documentation](https://help.ubuntu.com/community/UFW) if you need help configuring it.
+
+
+
+## Netplan configuration
+
+
+
+### Edit netplan configuration
+
+Edit netplan configuration file:  ```/etc/netplan/00-installer-config.yaml```
+  - Disable dhcp on NIC.
+  - Create two VLANs (40 and 41).
+  - Create three Bridges.
+    - br0: this is a bridge on the untagged VLAN1, and management interface of our server.
+    - br0-vlan40: this is a bridge on vlan40.
+    - br0-vlan41: this is a bridge on vlan41.
+  - Assign IPv4 address to br0.
+  - Assign IPv4 address to br0-vlan40.
+  - Assign IPv4 address to br0-vlan41.
+  - Configure routes.
+  - Configure DNS.
+
+
+The configuration file should look like this:
+
+```
+# network configuration:
+# eno1 - untagged vlan1
+# eno1-vlan40 - vlan interface to connect to tagged vlan40
+# eno1-vlan41 - vlan interface to connect to tagged vlan41
+# br0 - bridge for interface eno1 on untagged vlan1
+# br0-vlan40 - bridge on tagged vlan40
+# br0-vlan41 - bridge on tagged vlan41
+
+network:
+  version: 2
+  ethernets:
+    eno1:
+      dhcp4: false
+  vlans:
+    eno1-vlan40:
+      id: 40
+      link: eno1
+    eno1-vlan41:
+      id: 41
+      link: eno1
+  bridges:
+    br0:
+      interfaces: [eno1]
+      dhcp4: false
+      addresses: [192.168.150.1/24]
+      routes:
+        - to: default
+          via: 192.168.150.254
+          metric: 100
+          on-link: true
+      nameservers:
+        addresses: [1.1.1.1, 8.8.8.8]
+        search: []
+    br0-vlan40:
+      interfaces: [eno1-vlan40]
+      dhcp4: false
+      routes:
+        - to: 0.0.0.0
```

Comment:
I think this needs to be a network, not just an IP. E.g. `0.0.0.0/0` (or just "default"). Potentially, you need a different metric here to avoid conflicts with the default route above.

Same applies to the br0-vlan41 route below.

Reply:
This syntax seems to be accepted by Netplan. So nevermind, assuming the scenario worked for you correctly.